### DISCUSSION TITLE
How to build a LocalRequest client in AdHoc::on_liftoff？

### DISCUSSION THREAD

**Discussion by knota-lab** [rwf2/Rocket#1911] [Questions]
Rocket version：`rocket = {version = "0.5.0-rc.1", features = ["secrets", "tls", "json"]}`

```rust
AdHoc::on_liftoff("nacos", |r_orbit| {
    let client = Client::tracked(*r_orbit).unwrap();
}
```

I get this error.
```
cannot move out of `*r_orbit` which is behind a shared reference
move occurs because `*r_orbit` has type `Rocket<Orbit>`, which does not implement the `Copy` trait
```

**Comment by jebrosen** [comment]
Yes, this does not work. The purpose of `Client` is to dispatch "local" (in-memory) requests, such as in tests, *instead* of launching the server and listening for incoming connections.

What is your goal with creating a `Client` in `on_liftoff`?

**Reply by knota-lab** [reply]
I want to send the request to a third-party service to report current service info. I currently use `reqwest` to do this.
